helix-core/src/test/java/org/apache/helix/integration/controller/TestControllerLeadershipChange.java

```diff
@@ -74,6 +82,51 @@ public void afterClass() {
   }
 
   @Test
+  public void testControllerCleanUpClusterConfig() {
```

Comment:
there are 3 testcases: 1. we don't have DISABLED_INSTANCES at all. before the pipeline and after the pipeline config should match as no update should have happened. 2. this test scenario.   3. original DISABLE_INSTANCE related test-case verifying that it remains same.

Reply:
TFTR. I will add these 2 test cases after  #2134 is merged. (otherwise it is hard to test when we still have batch mode enabled and we write to DISABLED_INSTANCES)

Reply:
makes sense.. but once #2134 is checked-in, you can add.

Reply:
Added test and rebased.